apis/python/src/tiledbsoma/soma_dataframe.py

```diff
@@ -285,22 +262,17 @@ def write(self, values: pa.Table) -> None:
                     )
```

Comment:
Line 260 above - why convert using `to_pandas()` and then call `np.array`?  

`to_pandas` will always return a np.array as long as the type can be stored in NumPy (ie, isn't a Pandas extension type).  And `np.array` will fail given one of those extension types.  

`np.array` also make sa copy by default, so you end up with a lot of extra memory copies here.

I suspect you just want:   `values.column(name).to_pands()`

If you are concerned that your type mapper might return a type that can't be expressed in numpy, you could assert that the return value is instanceof np.array.

Reply:
I should add:   `values.column(name)` returns a `ChunkedArray`.  The latest PyArrow `ChunkedArray` also contains a `to_numpy()` method - maybe just use that to make your assumptions more explicit, and skip the copy?

Reply:
@bkmartinjr that predates this PR -- is not new to it or modified by it -- i don't even think i wrote this bit -- i'm happy to update on this PR if we want to go there

Reply:
I understand.  But a code review is a code review :-)

If you are asking me to file an issue, just let me know.

Reply:
good call @bkmartinjr -- i think the `to_pandas()` with also `np.array` is because without the later

```
TypeError: Cannot convert Series to numpy.ndarray
```

you are correct about `to_numpy()`:

```
>>> type(table.column('i'))
<class 'pyarrow.lib.ChunkedArray'>
>>> type(table.column('i').to_pandas())
<class 'pandas.core.series.Series'>
>>> type(np.asarray(table.column('i').to_pandas()))
<class 'numpy.ndarray'>
>>> type(table.column('i').to_numpy())
<class 'numpy.ndarray'>
```

where `table` is a `pyarrow.Table`

Reply:
@bkmartinjr thanks --

the `to_pandas()` with also `np.array` is because without the latter

```
TypeError: Cannot convert Series to numpy.ndarray
```

since

```
>>> type(table.column('i'))
<class 'pyarrow.lib.ChunkedArray'>
>>> type(table.column('i').to_pandas())
<class 'pandas.core.series.Series'>
>>> type(np.asarray(table.column('i').to_pandas()))
<class 'numpy.ndarray'>
```

you're right, there is `.to_numpy()`, which _does_ produce a `numpy.ndarray`, but unlike `to_pandas()` it doesn't take the `types_mapper` argument we need

in any event, that bit predates this PR -- is not new to it or modified by it -- my preference is to
take these (very valid!) copy-reduction concerns into a separately tracked issue/PR

Reply:
@bkmartinjr since these are very valid concerns, outside of what this PR changed, i'll file a follow-up issue to track

Reply:
#469